src/System.Numerics.Vectors/src/System/Numerics/Vector.cs

```diff
@@ -1288,89 +1287,25 @@ public bool Equals(Vector<T> other)
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(Int16))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashHelpers.Combine(hash, ((Int16)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(UInt32))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashHelpers.Combine(hash, ((UInt32)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(Int32))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashHelpers.Combine(hash, ((Int32)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(UInt64))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashHelpers.Combine(hash, ((UInt64)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(Int64))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashHelpers.Combine(hash, ((Int64)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(Single))
-                {
-                    for (int g = 0; g < Count; g++)
-                    {
-                        hash = HashHelpers.Combine(hash, ((Single)(object)this[g]).GetHashCode());
-                    }
-                    return hash;
-                }
-                else if (typeof(T) == typeof(Double))
-                {
+                if (typeof(T) == typeof(Byte) ||
+                    typeof(T) == typeof(SByte) ||
+                    typeof(T) == typeof(UInt16) ||
+                    typeof(T) == typeof(Int16) ||
+                    typeof(T) == typeof(UInt32) ||
+                    typeof(T) == typeof(Int32) ||
+                    typeof(T) == typeof(UInt64) ||
+                    typeof(T) == typeof(Int64) ||
+                    typeof(T) == typeof(Single) ||
+                    typeof(T) == typeof(Double))
+                {
+                    var hashCode = new HashCode();
                     for (int g = 0; g < Count; g++)
                     {
-                        hash = HashHelpers.Combine(hash, ((Double)(object)this[g]).GetHashCode());
+                        hashCode.Add(((T)(object)this[g]).GetHashCode());
```

Comment:
Is the cast still necessary here?  Same goes for the call to `GetHashCode`.
Can't this just be 

`hashCode.Add(this[g]);`